x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycle.java

```diff
@@ -187,7 +193,7 @@ protected Clock getClock() {
         ilmHistoryStore.set(new ILMHistoryStore(settings, new OriginSettingClient(client, INDEX_LIFECYCLE_ORIGIN),
             clusterService, threadPool));
         indexLifecycleInitialisationService.set(new IndexLifecycleService(settings, client, clusterService, threadPool,
-            getClock(), System::currentTimeMillis, xContentRegistry, ilmHistoryStore.get()));
+            getClock(), System::currentTimeMillis, xContentRegistry, ilmHistoryStore.get(), getLicenseState()));
```

Comment:
I believe not, since `XPackPlugin.licenseState` is a `SetOnce` object. Also, `XPackLicenseState` has an `update` method, which changes the `status` and is invoked from `LicenseService`. Must admit I am also not too familiar with the details, but I do believe this does not change behavior "in production".